homeassistant/components/sensor/filter.py

```diff
@@ -229,11 +293,11 @@ def _filter_state(self, new_state):
 
     def filter_state(self, new_state):
         """Implement a common interface for filters."""
-        filtered = self._filter_state(new_state)
-        if isinstance(filtered, Number):
-            filtered = round(float(filtered), self.precision)
-        self.states.append(filtered)
-        return filtered
+        filtered = self._filter_state(FilterState(new_state))
+        filtered.set_precision(self.precision)
+        self.states.append(copy(filtered))
+        new_state.state = filtered.state
```

Comment:
What is `new_state` here? is that a state object from Home Assistant?

Reply:
Because in that case you're not allowed to assign to it.

Reply:
no it is a FilterState object.